addons/utm/models/utm_mixin.py

```diff
@@ -35,9 +38,12 @@ def default_get(self, fields):
                 # if we receive a string for a many2one, we search/create the id
                 if field.type == 'many2one' and isinstance(value, str) and value:
                     Model = self.env[field.comodel_name]
-                    records = Model.search([('name', '=', value)], limit=1)
+                    # also search for achieved records for campaign_id
+                    records = Model.with_context(active_test=not field_name == 'campaign_id').search([('name', '=', value)], limit=1)
```

Comment:
Can you rename `records` to `record` while you're at it?

It's very confusing to me.